File: Python/Fibonacci.py
#!/usr/bin/env python
import sys

# Author: Andrew Jarombek
# Date: 7/15/2016
# Enter a number and have the program generate the Fibonacci sequence to that
# number or to the Nth number.

class Fibonacci:
    
    fibs = [0,1]
    
    def fibSequence(self, length):
        # Make sure user is trying to find a positive fibonacci number
        if length < 0:
            sys.stderr.write("Usage: %s [fibLength] (length must be positive)\n" % sys.argv[0])
            sys.exit(1)
        
        if length is 0:
            return self.fibs[:0]
        
        for index in range(1, length):
            self.fibs.append(self.fibs[index-1] + self.fibs[index])
        return self.fibs[1:]

if __name__ == '__main__':
    
    # If no length is specified display an error message and exit
    if not sys.argv[1:]:
        sys.stderr.write("Usage: %s [fibLength]\n" % sys.argv[0])
        sys.exit(1)
    
    seqLength = int(sys.argv[1])
        
    f = Fibonacci()
    print("%s\n" % str(f.fibSequence(seqLength)))

File: Ruby/fibonacci.rb
# Author: Andrew Jarombek
# Date: 7/19/2016
# Enter a number and have the program generate the Fibonacci sequence to that number or to the Nth number.

class Fibonacci
	# Instance variable for the fibonacci sequence array
	attr_accessor :sequence

	def gen_fibonacci(n)
		if n <= 0
			puts "The ##{n} fibonacci number is #{sequence[0]}"
			return sequence[0]
		elsif n == 1
			puts "The ##{n} fibonacci number is #{sequence[1]}"
			return sequence[1]
		else
			# Similar to a for loop in other languages
			(n-1).times do |i|
				# adds the value of the expression to the end of the array 
				sequence << (sequence[i+1] + sequence[i])
			end
		end
		puts "The ##{n} fibonacci number is #{sequence[n]}"
		return sequence[n]
	end
end

fibnumber = Integer(ARGV[0])
f = Fibonacci.new
# Initialize the instance variables
f.sequence = [0, 1]
f.gen_fibonacci(fibnumber)
